### PR TITLE
doc: Document how to replace packages and menu items

### PR DESCRIPTION
**Pull request by stefwalter**
This should help people who want to replace certain functionality
of Cockpit when shipping their own parts.

### CODE REVIEW

**Comment by stefwalter** [comment]
@ smirnovigor We've added the functionality that you needed with replacing menu items. Here I've documented it better. Does it solve the same problem you were trying to solve in #4580 and reported in #4558

**Comment by stefwalter** [comment]
CC @smirnovigor

File: doc/guide/packages.xml [hunk 1]
```diff
@@ -253,4 +253,50 @@ mypackage/test.min.js.gz
 
   </section>
 
+  <section id="package-replace">
+    <title>Replacing an existing package</title>
+
+    <para>If the functionality in a package replaces one of the default packages
```
Comment (dperpeet) [inline doc/guide/packages.xml]: This relates to all packages, right? Not just `default` packages, whatever those are. I suggest ... `replaces that of another package`

File: doc/guide/packages.xml [hunk 2]
```diff
@@ -253,4 +253,50 @@ mypackage/test.min.js.gz
...
+      then it can replace that package by claiming the same <code>name</code> and a
+      higher <code>priority</code>.</para>
+
+    <para>For example a package in the <filename>/usr/share/cockpit/disks</filename>
+      directory should replace the <emphasis>storage</emphasis> package that comes
+      with Cockpit would have a <filename>manifest.json</filename> that looks like this:</para>
```
Comment (dperpeet) [inline doc/guide/packages.xml]: Sentence is a bit misleading.
```
For example, a package in the <filename>/usr/share/cockpit/disks</filename>
directory could replace Cockpit's <emphasis>storage</emphasis> package with
a <filename>manifest.json</filename> like this:
```

File: doc/guide/packages.xml [hunk 3]
```diff
@@ -253,4 +253,50 @@ mypackage/test.min.js.gz
...
+  }
+}
+</programlisting>
+
+    <para>It is also possible to hide or change labels on the menu items of an existing
+      package by including a <filename>override.json</filename> in that existing package's
+      directory.</para>
+
+    <para>For example an <filename>/usr/share/cockpit/systemd/override.json</filename> could
+      hide the <emphasis>Logs</emphasis> and move the <emphasis>Services</emphasis> menu item
```
Comment (dperpeet) [inline doc/guide/packages.xml]: I suggest either `the <emphasis>Logs</emphasis> item` (or `entry`) or without `the`

Comment (stefwalter) [inline doc/guide/packages.xml]: Fixed.

File: doc/guide/packages.xml (hunk 2 above)
Comment (stefwalter) [inline doc/guide/packages.xml]: Used this wording.

File: doc/guide/packages.xml (hunk 1 above)
Comment (stefwalter) [inline doc/guide/packages.xml]: Used this wording.

**Comment by igorSmirnovPlexi** [comment]
@stefwalter Yes, it exactly what I need. Waiting to check this out :)

**Pull request merged**